### DISCUSSION TITLE
Modify Context value for consecutive resolvers

### DISCUSSION THREAD

**Discussion by mezykr-oanda** [apollographql/apollo-server#5224] [Q&A]
Hey, I'm trying to find a way to change/set some value in context (or other place) to make it available in nested consecutive children resolvers. I know that I can propagate a value from parent resolver down but In my case I would have to do that by several layers (10+)... Setting this value by operation context method will not work - this is something what I can set basing on value returned from some top-level resolver ... Please see below diagram for reference. Thank you for any help on this ...
![Untitled drawing](https://user-images.githubusercontent.com/63653601/118657046-64b96000-b7eb-11eb-8341-1baa432e6f3a.png)

**Comment by glasser** [comment]
Apollo Server doesn't have a great answer for this today; the `context` is encouraged to be constant across resolvers.

I think graphql-java's [local context](https://www.graphql-java.com/blog/deep-dive-data-fetcher-results/) is a good design (though it only passes down one level at a time).
